src/components/ItemsForSale.js

```diff
@@ -0,0 +1,179 @@
+import React, { Component } from 'react'
+import { Card, Col, Row } from 'reactstrap'
+import { HicetnuncContext } from '../context/HicetnuncContext'
+import Menu from '../components/Menu'
+import { LazyLoadImage } from 'react-lazy-load-image-component'
+import InfiniteScroll from 'react-infinite-scroll-component'
+import { BabelLoading } from 'react-loadingg'
+
+export default class ItemsForSale extends Component {
+  static contextType = HicetnuncContext
+
+  collect = (event, index) => {
+    if (this.context.Tezos == null) {
+      alert('sync')
+    } else {
+      this.context.collect(
+        1,
+        this.props.items[this.props.index].swaps[0].swap_id,
+        this.props.items[this.props.index].swaps[0].xtz_per_objkt * 1
+      )
+    }
+  }
+
+  render() {
+    const { index } = this.props
+    const info = (index) => (
+      <div style={{ display: 'inline' }}>
+        <a
+          style={{
+            color: '#000',
+            '&:hover': {
+              color: '#000',
+            },
+          }}
+          href={`/objkt/${this.props.items[index].token_id}`}
+        >
+          OBJKT#{this.props.items[index].token_id}
+        </a>
+      </div>
+    )
+    const image = (index) => (
```

Comment:
I'm slowly trying to remove duplicated code from all the pages. I've only had time to do src/pages/objkt-display
But on that page you can see that there's a renderer that renders the correct media based on the mimetype. We should use that, and avoid having duplicates of image/video components everywhere. This is because we want to introduce more media types in the future, and we dont want to keep duplicating components around. Can you update so it uses the "renderMediaType" (https://github.com/hicetnunc2000/hicetnunc/blob/master/src/components/media-types/index.js)